src/components/ServiceHealth/HealthIndicator.tsx

```diff
@@ -28,8 +28,8 @@ export class HealthIndicator extends React.PureComponent<Props, {}> {
     this.updateHealth(props.health);
   }
 
-  componentWillReceiveProps(nextProps: Props) {
-    this.updateHealth(nextProps.health);
+  componentDidUpdate() {
```

Comment:
Why replace `componentWillReceiveProps` with `componentDidUpdate` According to the React blog on [Component Lifecycle changes](https://reactjs.org/blog/2018/03/29/react-v-16-3.html#component-lifecycle-changes) it should be replaced with `getDerivedStateFromProps`
I have concerns with `componentDidUpdate` as it happens after the `render` method where the `componentWillReceiveProps` happens before `render`
See Diagram:
![reactjs-components-life-cycle2](https://user-images.githubusercontent.com/1312165/39591744-7d811180-4eb9-11e8-85f0-d2e58cf9dd2d.png)

Reply:
@vnugent what do you think here?

Reply:
I comment his in the JIRA, the point is that you can't call another method from a static method so or we make the fetch inside this static method or we need to use another lifecycle method @mtho11, I read this in the doc. Maybe I am wrong but IDE and debugger don't let me to do it ^^

Reply:
I just looked at the previous version with  `componentWillReceiveProps()`.  If all `updateHealth()` does is computing various new values for render(), `updateHealth()` can just be called in `componentDidMount()`.  

For future improvement I'd change `updateHealth(Health)` to arrow function to make it really clear what `this` is referring to, and break it up for readability.  Without knowing what Health argument is, it'd be hard for someone else to debug the function.